NT undercover changes

TDSSS opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
This PR effectively renames the existing 'NT Undercover Operative' into the newly created 'NT Undercover Officer', then creates a replacement NT Undercover Agent with reduced gear and tools.

The idea is that the Agent is an outfit that can be given to ghosts during events or in response to threats as an alternative to sending an ERT, while the Undercover Officer maintains access to all the OP admin-only tools that would be overkill to give to a player.

Detailed breakdown of changes:
1. Combat gloves of both are replaced with insulated gloves. Combat gloves are not exactly common and draw unwanted attention, an undercover operative should IMO not be using them. The heat resistance is only rarely relevant.
2. An NT Saber SMG was added to both, together with 3 magazines and a suppressor. Since the Agent loses access to the CC combat implant, he needed a replacement weapon. The Saber SMG is deadly, small, suppressed, made by NT and was used by NT Undercover operatives in the past. The Undercover Officer gets this too but can freely ditch it at CC if he doesn't want it, to help his cover.
3. As mentioned, Agents lose the CC combat implant which had a pulse pistol, mindshield implanter, baton, omni-door remote and healing injector. They keep welding shield, nutritien implant, sec hud implant and gain a baton implant for unarmed combat.  They also still have a dust implant
4. Agents no longer have full Centcom access. They now have the same access as an ert team leader, ie basic centcom access. Mostly a flavour thing but it is a role intended to be given to players, so better be safe here.
5. Agents also get an emp flashlight, for disabling coms or borgs or other misc utility.


## Why It's Good For The Game
NT Undercover Operatives have a lot of potential, IMO. Sometimes, CC wants someone eliminated quietly. While an admin could spawn in as the old version of undercover operative, that is in many cases suboptimal since it 1) is pretty damn unfair, raising the barrier for using it quite high. Good luck escaping someone with a centcom combat implant 2) Giving ghosts something to do can't hurt.

So I intend to give us another tool in our event/response toolbox. Maybe rather than an ert a few undercover agents can be sent to help deal with a situation like a mutiny discreetly, giving ghosts something to do and the antagonists still a fair fighting chance. And if full admeme force is required, the Undercover Officer still has all the good stuff.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Images of changes
<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif of your feature if you want -->

## Changelog
:cl:
add: NT Undercover Officers
tweak: Changes to NT Undercover Operatives, renamed to NT Undercover Agent
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Mitchs98 commented:
https://www.youtube.com/watch?v=6iaR3WO71j4

Relevant and obligatory.

Kyep commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -220,24 +220,40 @@
 
 /datum/outfit/admin/nt_undercover
 	name = "NT Undercover Operative"
```
Maybe if you're creating a lesser version, use 'Agent' instead of 'Operative'.
Rationale is that the syndicate already have a similar structure ("syndicate agents" are undercover players on station, "syndicate officers" are admin-only event roles). So it makes it easier to understand what each does, and the difference between them. Using 'Agent' instead of officer also prevents both of them having the same abbreviation ("NTUO").

TDSSS replied:
I didn't think about the abbreviation thing, will go ahead and change it.

Kyep commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -220,24 +220,40 @@
 	back = /obj/item/storage/backpack
 	belt = /obj/item/storage/belt/utility/full/multitool
-	gloves = /obj/item/clothing/gloves/combat
+	gloves = /obj/item/clothing/gloves/color/yellow
```
Is this because combat gloves do things that insulated gloves don't?
Or is this just to help them blend in more?

ZomgPonies replied:
Blend in more. Combat gloves also offer heat protection, but that is so rarely used it's not a factor.

TDSSS replied:
It's for blending in more. Combat gloves are fairly rare and the fire resistance rarely comes up, imo. Nobody bats an eye at am assistant with insulated gloves.

Kyep commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -220,24 +220,40 @@
-		/obj/item/pinpointer/crew = 1
+		/obj/item/pinpointer/crew = 1,
+		/obj/item/gun/projectile/automatic/proto = 1,
+		/obj/item/ammo_box/magazine/smgm9mm = 3,
```
You might also want to give them a comms jammer, or something so that they don't instantly get dogpiled by sec when their subject screams on radio.

TDSSS replied:
Good idea. I thought about giving them an emp flashlight or something but that works, too.

Qwertytoforty replied:
Radio jammer may not work, due to the fact they can still call out on comns and make a lot of radio noise jibberish, which will probably cause the AI to track them, and people nearby may notice the radio jammer and call sec. EMP flashlight or a muting chemical would probably work better.

Kyep replied:
Could make a modified version of radio jammer that totally silences radio, instead of just garbling it.

Kyep replied:
Also, they only have one spare mag. Once that's gone, they're stuffed.

TDSSS replied:
Uh, no they have 3 spare mags. If 3 spare mags and the one in the gun isn't enough to do the job it's less an undercover op and more a war.

Kyep replied:
Oh, =3. Nevermind on that one.

TDSSS commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -220,24 +220,40 @@
 	implants = list(
 		/obj/item/implant/dust
 	)
+	cybernetic_implants = list(
```
Good idea!

TDSSS commented:
It is definitely a role where you need to be careful so it doesn't lead to things going out of control, ie giving ghosts clear guidance on what they are allowed to do. I think dusting would generally be correct when caught by sec. As for keeping people dead, I assumed the operative would have to get creative or bring the body back to CC, but we could explore alternatives.

>This new version could easily be spawned from Smite->Hunter, so they'd get a pinpointer leading to their target (you've verified that works, right?).

I'm not sure what I should verify. I have not tried making hunter use/spawn this outfit, I just tested it with reincarnate and with the equip verb.

Kyep commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -219,25 +219,41 @@
+	// Ghost role for undercover antics on behalf of NT. Much weaker than the "NT Undercover Officer" below.
+	//Lacks CC implant and CC access. Default weapon is silenced saber SMG.
 
 	uniform = /obj/item/clothing/under/color/black
```
Maybe use under/color/random instead of under/color/black
Otherwise, if an admin spawns 5 of these, they'll all appear in the exact same black outfit, which makes them super-easy to identify once you know what to look for.

TDSSS commented:
Addressed a few comments. To sum up:
Changed some comment
Changed outfit to random colour
Renamed from operative to agent
Given sec HUD
Flashlight turned into emp flashlight

An emp flashlight seems more universally useful and can help out against rogue borgs or AIs, traitors with 5 energy weapons in their backpack and implants, etc.

Denthamos commented:
Personally I never see myself giving a normal player this role or using it.  If I need something handled from the CC side of things I will be doing it myself and not handing that off to a player which could make things worse.

The only thing CC wise that normal players should ever be in control with is ERT's and NT Rep/blueshield.  CC and syndicate officers/high command both should always be admin only.

TDSSS commented:
@Denthamos 

Nobody says it's an outfit you'll see used _often_. We have lots of outfits that are only very rarely used, like Singuloth Knight, Chrono Legionnaire, etc.

> If I need something handled from the CC side of things I will be doing it myself and not handing that off to a player which could make things worse.

I think you're misunderstanding the role this outfit should serve, there. It's not intended to flawlessly _solve_ problems. It's intended to give the problem it's supposed to solve (ie rampaging vampire, traitor arranging a mutiny, etc.) a fair fighting chance while attacking from a different angle than an ERT. 

Sure, you can spawn yourself in as an NT Undercover Operative and play fair and not make use of your full range of gear, but the players you're fighting will know you were holding back, that any win they got was only because you allowed it. A victory over another player that's given limited tools and giving his best is much more rewarding.

> The only thing CC wise that normal players should ever be in control with is ERT's and NT Rep/blueshield. CC and syndicate officers/high command both should always be admin only.

How is this so much different from trusting a ghost with being an ERT? The role comes with no official authority at all, wheras an ert can easily overrule the captain or execute people if they deem it necessary. It is a 'thing CC wise' only in that it will mostly be sent from CC, and not even that is a given.

Denthamos commented:
> @Denthamos
> 
> Nobody says it's an outfit you'll see used _often_. We have lots of outfits that are only very rarely used, like Singuloth Knight, Chrono Legionnaire, etc.
> 

Because we do not need more things bloating up the code and clogging menus.  
> > If I need something handled from the CC side of things I will be doing it myself and not handing that off to a player which could make things worse.
> 
> I think you're misunderstanding the role this outfit should serve, there. It's not intended to flawlessly _solve_ problems. It's intended to give the problem it's supposed to solve (ie rampaging vampire, traitor arranging a mutiny, etc.) a fair fighting chance while attacking from a different angle than an ERT.
> 

I think you are misunderstanding the staffs stance on how we prefer to solve such issues in game.  Its a long standing knowledge that anything that comes from central command is admin related and we do not need non staff in such roles that will cause further confusion for players.  We already have issues with the blueshield and rep which still come up now and then which makes those roles (especially blueshield) very controversial.

> Sure, you can spawn yourself in as an NT Undercover Operative and play fair and not make use of your full range of gear, but the players you're fighting will know you were holding back, that any win they got was only because you allowed it. A victory over another player that's given limited tools and giving his best is much more rewarding.
> 
> > The only thing CC wise that normal players should ever be in control with is ERT's and NT Rep/blueshield. CC and syndicate officers/high command both should always be admin only.
> 
> How is this so much different from trusting a ghost with being an ERT? The role comes with no official authority at all, wheras an ert can easily overrule the captain or execute people if they deem it necessary. It is a 'thing CC wise' only in that it will mostly be sent from CC, and not even that is a given.

Because you are essentially giving a player a role from central command that does antagonist actions against the crew.  NTUO is used very rarely for a reason because it muddies up the water on which factions are antagonistic and which are not and is like using a scalpel to cut a wound open.  What you propose here would be like using a machete for the same use and not what we need.
